### PR TITLE
Remove Buttons from Main Menu with Arcade Feature or WASM32 Target

### LINKED ISSUE #180: Remove Exit Button from Main Menu for Arcade Build
Remove ability for users to exit the game from the main menu when playing the arcade build.

### PR DESCRIPTION
**Pull request by cdsupina**
We don't want users to be able to quit the game from the main menu when it is running on an arcade machine. Removed all other buttons besides the Play Game button for the arcade build. Fixes #180

### CODE REVIEW

File: src/ui/main_menu/button.rs [hunk 1]
```diff
@@ -92,7 +92,13 @@ pub(super) trait UiChildBuilderExt {
 
 impl UiChildBuilderExt for ChildBuilder<'_> {
     fn spawn_main_menu_buttons(&mut self, ui_assets: &UiAssets, font: Handle<Font>) -> &mut Self {
-        for action in MAIN_MENU_BUTTON_ORDER.iter() {
+        let buttons = if cfg!(feature = "arcade") {
+            vec![MainMenuButtonActionComponent::EnterInstructions]
+        } else {
+            Vec::from(MAIN_MENU_BUTTON_ORDER)
+        };
```
Comment (varoonp123) [inline src/ui/main_menu/button.rs]: Im busy this week, so ill test this over the weekend. But I think this might not be right. To know the currently selected button when you press up/down with the arrows/gamepad, we [increment/decrement a counter](https://github.com/thetawavegame/thetawave/blob/acd61c93a237569e33ea9f921850e34a3977b486/src/ui/main_menu/button.rs#L274) whenever you press up/down and mod by ` MAIN_MENU_BUTTON_ORDER.len()`. Then we [index into `MAIN_MENU_BUTTON_ORDER`](https://github.com/thetawavegame/thetawave/blob/acd61c93a237569e33ea9f921850e34a3977b486/src/ui/main_menu/button.rs#L296) to determine the event to send/action to take. So this might lead to unexpected behavior (the Compendium button is still there, even though the UI doesnt show it. Remember that this ChildBuilder is just used to show what is happening in an internal `Local<MainMenuUIState>` ([basically an integer](https://github.com/thetawavegame/thetawave/blob/acd61c93a237569e33ea9f921850e34a3977b486/src/ui/main_menu/button.rs#L169)). It isn't really **the** "button) .


So _really_ you either
1. Make this generic over the ActionEventEnum+ActionEventEnum handling system+array of values that correspond to those actions
2. At compile time, set `MAIN_MENU_BUTTON_ORDER` to be a singleton array when we use the arcade feature
3. Toggle different main menu setups by placing each implementation in its own plugin and enabling/disabling those plugins in `UIPlugin::build`, and even based in  CLI parameters by making `UIPlugin`  have a field like `simplified_start_menu: bool` or `main_menu_kind: MainMenuKindEnum` .  Something like: rename `MainMenuPlugin` to `StackedButtonsMainMenuImplPlugin` and the `main_menu` module to `stacked_buttons_main_menu` . Then do something like the following in `MainMenuPlugin::build`
```rust

// It is important that exactly one of these branches is taken because the single plugin on each branch
// provides some way for the user to go from `AppStates::MainMenu` to `AppStates::Instructions` and 
// gets to assume that it "controls the world" as far as drawing on the screen goes, if and only if we are 
// in `AppStates::MainMenu`. This invariant cannot be easily expressed in a type system
if cfg!(feature=arcade)` // || self.simple_main_menu
{
    self.add_plugins(ArcadeSinglePressMainMenuImplPlugin);
}
else {
self.add_plugins(StackedButtonsMainMenuImplPlugin);

}
```

Basically, if all you can do is go to the instructions screen, I dont think you even need a button. Make the UI you _actually_ want in the arcade and dispatch to that.

**Comment by cdsupina** [comment]
@LordDeatHunter pointed out that we should also disable the exit button with the wasm build

File: src/ui/main_menu/button.rs (hunk 1 above)
Comment (cdsupina) [reply src/ui/main_menu/button.rs]: I chose to do option 2. Once we have more content to put on the main menu, like high scores, I think it will warrants its own plugin.
